components/brave_wallet/browser/simulation_response_parser.cc

```diff
@@ -517,6 +826,219 @@ mojom::BlowfishSolanaStateChangeRawInfoPtr ParseStateChangeRawInfo(
+  } else if (error == "ACCOUNT_LOADED_TWICE") {
+    return mojom::BlowfishSolanaErrorKind::kAccountLoadedTwice;
+  } else if (error == "ACCOUNT_NOT_ASSOCIATED_WITH_THIS_MINT") {
+    return mojom::BlowfishSolanaErrorKind::kAccountNotAssociatedWithThisMint;
+  } else if (error ==
+             "ADVANCING_STORED_NONCE_REQUIRES_A_POPULATED_RECENTBLOCKHASHES_"
+             "SYSVAR") {
+    return mojom::BlowfishSolanaErrorKind::
+        kAdvancingStoredNonceRequiresAPopulatedRecentblockhashesSysvar;
+  } else if (error == "ALREADY_IN_USE") {
+    return mojom::BlowfishSolanaErrorKind::kAlreadyInUse;
+  } else if (error == "AN_ACCOUNT_WITH_THE_SAME_ADDRESS_ALREADY_EXISTS") {
+    return mojom::BlowfishSolanaErrorKind::
+        kAnAccountWithTheSameAddressAlreadyExists;
+  } else if (error ==
+             "ATTEMPT_TO_DEBIT_AN_ACCOUNT_BUT_FOUND_NO_RECORD_OF_A_PRIOR_"
+             "CREDIT") {
+    return mojom::BlowfishSolanaErrorKind::
+        kAttemptToDebitAnAccountButFoundNoRecordOfAPriorCredit;
+  } else if (error == "ATTEMPT_TO_LOAD_A_PROGRAM_THAT_DOES_NOT_EXIST") {
+    return mojom::BlowfishSolanaErrorKind::
+        kAttemptToLoadAProgramThatDoesNotExist;
+  } else if (error == "BAD_REQUEST") {
+    return mojom::BlowfishSolanaErrorKind::kBadRequest;
+  } else if (error == "BLOCKHASH_NOT_FOUND") {
+    return mojom::BlowfishSolanaErrorKind::kBlockhashNotFound;
+  } else if (error == "CANNOT_ALLOCATE_ACCOUNT_DATA_OF_THIS_LENGTH") {
+    return mojom::BlowfishSolanaErrorKind::
+        kCannotAllocateAccountDataOfThisLength;
+  } else if (error == "CANNOT_ASSIGN_ACCOUNT_TO_THIS_PROGRAM_ID") {
+    return mojom::BlowfishSolanaErrorKind::kCannotAssignAccountToThisProgramId;
+  } else if (error == "FIXED_SUPPLY") {
+    return mojom::BlowfishSolanaErrorKind::kFixedSupply;
+  } else if (error == "INSTRUCTION_DOES_NOT_SUPPORT_NATIVE_TOKENS") {
+    return mojom::BlowfishSolanaErrorKind::
+        kInstructionDoesNotSupportNativeTokens;
+  } else if (error == "INSTRUCTION_DOES_NOT_SUPPORT_NON-NATIVE_TOKENS") {
+    return mojom::BlowfishSolanaErrorKind::
+        kInstructionDoesNotSupportNonNativeTokens;
+  } else if (error == "INSUFFICIENT_FUNDS") {
+    return mojom::BlowfishSolanaErrorKind::kInsufficientFunds;
+  } else if (error == "INSUFFICIENT_FUNDS_FOR_FEE") {
+    return mojom::BlowfishSolanaErrorKind::kInsufficientFundsForFee;
+  } else if (error == "INVALID_INSTRUCTION") {
+    return mojom::BlowfishSolanaErrorKind::kInvalidInstruction;
+  } else if (error == "INVALID_MINT") {
+    return mojom::BlowfishSolanaErrorKind::kInvalidMint;
+  } else if (error == "INVALID_NUMBER_OF_PROVIDED_SIGNERS") {
+    return mojom::BlowfishSolanaErrorKind::kInvalidNumberOfProvidedSigners;
+  } else if (error == "INVALID_NUMBER_OF_REQUIRED_SIGNERS") {
+    return mojom::BlowfishSolanaErrorKind::kInvalidNumberOfRequiredSigners;
+  } else if (error == "LAMPORT_BALANCE_BELOW_RENT-EXEMPT_THRESHOLD") {
```

Comment:
`_RENT-EXEMPT_` Is it expected hyphen instead of underscore there? 
It would be helpful to have some public link to spec on these strings parsers if possible.

Reply:
The hyphen is used in the official documentation.

Can't link directly to the list content but here is the documentation page that contains the list of errors:

https://docs.blowfish.xyz/v2023-03-08/reference/scan-transactions-solana

From there, under "RESPONSES", click "200" -> "ERROR OBJECT | NULL"

The "kind" section contains the list of error kinds for Solana Transaction scans.